### PR TITLE
file_system: detect procfs accessibility before using it

### LINKED ISSUE #13116: proc file read failure flooding logs
### What version of TiKV are you using?

N/A

### What operating system and CPU are you using?

N/A

### Steps to reproduce

N/A

### What did happened?

[WARN] [[proc.rs:65](http://proc.rs:65/)] ["failed to open proc file: No such file or directory (os error 2)"]

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Close #13116

What's Changed:

```commit-message
None
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/file_system/src/io_stats/proc.rs [hunk 1]
```diff
@@ -140,14 +129,17 @@ impl AtomicIOBytes {
...
         GLOBAL_IO_STATS[sentinel.io_type as usize]
             .fetch_add(io_bytes - sentinel.last_flushed, Ordering::Relaxed);
         sentinel.last_flushed = io_bytes;
     }
 }
 
 pub fn init() -> Result<(), String> {
+    ThreadID::current()
+        .fetch_io_bytes()
+        .map_err(|e| format!("failed to fetch I/O bytes from proc: {}", e))?;
```
Comment (5kbpers) [inline components/file_system/src/io_stats/proc.rs:142]: Should we log a warning here?

Comment (tabokie) [reply components/file_system/src/io_stats/proc.rs:142]: It will warn in `server::init_io_utility`.

**Review by Connor1996 — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**